sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -43,11 +43,12 @@ struct joint_matrix {
   }
 };
 
-#ifdef __SYCL_DEVICE_ONLY__
+// #ifdef __SYCL_DEVICE_ONLY__
+// The wi_data class has to have a jm member so that we can compile code
+// that uses members of wi_element such as get_coord() in the host as well.
```

Comment:
So I think my first comment stands: are you sure you want to be able to call `get_coord` from host? Because I didn't think you would want this. Making this decision is leading you into trouble with follow on changes that are then leading to the cuda failures.

Reply:
We do not want get_coord() to be called from the host. But even though the call is in a kernel (which the device is supposed to execute,) there is no way to ensure that the compilation for host will be skipped right? So in that case we will have a compilation error.

Reply:
Sure I see. The `wi_data` overload that was constructed from an `marray` that you've removed here was added for the same reason so that host compilation would pass for both nvidia and intel backends. Basically I think you need to add this `get_coord` support in a way that doesn't break this.
I haven't thought it thorugh but I guess reverting the above mentioned change you made and overloading wi_element/get_coord for host compilation or similar could work. Or whatever you prefer so long as it allows the existing tests to pass basically.

Reply:
I added one solution @JackAKirk Let me know if this looks fine to you